appserver/data/data-core/src/main/java/fish/payara/jakarta/data/core/util/DataCommonOperationUtility.java

```diff
@@ -124,10 +124,26 @@ public static EntityManager getEntityManager(String applicationName) {
         ApplicationInfo applicationInfo = applicationRegistry.get(applicationName);
         List<EntityManagerFactory> factoryList = applicationInfo.getTransientAppMetaData(EntityManagerFactory.class.toString(), List.class);
         if (factoryList.size() == 1) {
-            EntityManagerFactory factory = factoryList.get(0);
+            EntityManagerFactory factory = factoryList.getFirst();
             return factory.createEntityManager();
         }
-        return null;
+
+        List<EntityManagerFactory> result = factoryList.stream().filter(factory -> {
+            Map<String, Object> properties = factory.getProperties();
+            if (properties == null || properties.isEmpty()) {
+                return false;
+            }
+            if (!properties.containsKey("fish.payara.jakarta.data.usePU")) {
```

Comment:
Not a blocker, but I'm not a fan of this name.
I'd drop the jakarta.

Also, since this is being used in 3 separate locations here - an enum or static final string for this property name seems appropriate.